protocol/common/message.go

```diff
@@ -504,7 +529,7 @@ func (m *Message) PrepareContent(identity string) error {
 	// Leave it set if already set, as sometimes we might run this without
 	// an identity
 	if !m.Mentioned {
-		m.Mentioned = visitor.mentioned
+		m.Mentioned = visitor.mentioned || containsMentionOf(m.Text, EveryoneMentionLiteral)
```

Comment:
Maybe there is a way to put `@everyone` in the checks done for `runMentionsAndLinksVisitor` but I have no idea how it works so it can be done later.

Reply:
Ah, that's what I did initially. The problem then is however, that when a node of the AST is visited, we don't know it's surroundings. These will end up in the `default` case (they aren't `ast.Mention` or `ast.Link`).

So the `literal` to inspect will be "@everyone" even though the message might be  [backtick]@everyone[backtick] at which point we don't want this to become recgonized as mention.

The options are to

1. either extend the markdown parser to know what an `@everyone` mention is and that it's to interpret as a mention or
2. Simply check the literal on the entire text (instead of AST)

I went with 2) as it seem simpler

Reply:
Well, now that I read https://github.com/status-im/status-go/pull/3026#issuecomment-1353427503 I guess I'll have to make it work with 1)  haha